src/main/java/org/jenkinsci/plugins/durabletask/BourneShellScript.java

```diff
@@ -98,6 +98,16 @@ public String getScript() {
         shf.write(s, "UTF-8");
         shf.chmod(0755);
 
+        scriptPath = shf.getRemote();
+        List<String> args = new ArrayList<String>();
+
+        if (!ws.act(new DarwinCheck())) { // JENKINS-25848
+            args.add("nohup");
+        }
+        if (ws.act(new WindowsCheck())) { // JENKINS-40255
```

Comment:
I guess you missed the point of my suggestion, which was to make a *single* Remoting call which check for both OS X and Windows, for example by returning an `enum`.